addon/components/frost-autocomplete-dropdown.js

```diff
@@ -0,0 +1,490 @@
+        if (!wrapLabels) {
+          const clonedTextElement = clonedTextElements[index]
+          const updatedData = trimLongDataInElement(clonedTextElement)
+
+          if (updatedData) {
+            textElement.textContent = updatedData.text
+            textElement.setAttribute('title', updatedData.tooltip)
+          }
+        }
+
+        if (filter) {
+          const pattern = new RegExp(filter.replace(/[.*+?^${}()|[\]\\]/, '\\$&'), 'i')
+          const textWithMatch = textElement.textContent.replace(pattern,
+            "<span class='frost-select-list-item-text-highlight'>$&</span>")
+
+          // If rendered text has changed, update it
+          if (textElement.innerHTML !== textWithMatch) {
+            textElement.innerHTML = textWithMatch
+          }
+        }
+      })
+    }
+
+    this._removeListItemEventListeners(dropdownListElement)
+
+    dropdownListElement.replaceWith(clonedDropdownListElement)
+
+    updateText(textElements, clonedTextElements)
+
+    clonedDropdownListElement.replaceWith(dropdownListElement)
+
+    this._addListItemEventListeners(dropdownListElement)
+
+    // Make sure we scroll back to where the user was
+    document.getElementById('frost-select-list').scrollTop = scrollTop
+  },
+
+  // == Tasks =================================================================
+
+  updateTask: task(function * () {
+    this._isUpdating = true
+
+    while (Date.now() - this._lastInteraction < 250) {
+      const $element = get(this.attrs, '$element.value') || get(this.attrs, '$element')
+
+      if ($element) {
+        const props = this._updatePosition($element)
+        this.setProperties(props)
+      }
+
+      yield timeout(FPS)
+    }
+
+    this._isUpdating = false
+  }),
+
+  // == DOM Events ============================================================
+
+  // == Lifecycle Hooks =======================================================
+
+  didReceiveAttrs () {
+    const $element = this.get('$element')
+    let props = {}
+
+    if ($element) {
+      props = merge(props, this._updatePosition($element))
+    }
+
+    const receivedHook = this.get('receivedHook')
+
+    if (receivedHook && receivedHook !== this.get('hook')) {
+      deprecate(
+        'receivedHook has been deprecated in favor of hook',
+        false,
+        {
+          id: 'receivedHook-deprecated',
+          until: '2.0.0'
+        }
+      )
+
+      props.hook = receivedHook
+    }
+
+    if (Object.keys(props).length !== 0) {
+      this.setProperties(props)
+    }
+  },
+
+  didInsertElement () {
+    $('.frost-autocomplete-dropdown .frost-text-input').focus() // Focus on filter
+
+    this._updateHandler = () => {
+      run(() => {
+        if (this.isDestroyed || this.isDestroying) return
+        this._lastInteraction = Date.now()
+
+        if (!this._isUpdating) {
+          this.get('updateTask').perform()
+        }
+      })
+    }
+
+    /* eslint-disable complexity */
```

Comment:
We should not be employing `eslint-disable complexity` in new code.  It should only be used to temporarily address existing code to work around additions or refactors but in new code it should just be written correctly.